src/views/toggle_button.rs

```diff
@@ -0,0 +1,151 @@
+use floem_reactive::{create_effect, RwSignal};
+use floem_renderer::Renderer;
+use kurbo::{Point, Size};
+use winit::keyboard::{Key, NamedKey};
+
+use crate::{id, prop, style_class, view::View};
+
+use super::Decorators;
+
+prop!(pub ToggleBg: Option<peniko::Color> {} = None);
+prop!(pub ToggleFg: Option<peniko::Color> {} = None);
```

Comment:
I'd add a builtin `Foreground foreground nocb: Option<Color> {} = None,` property and use that along with `Background`. They should be read in the `style` pass, see `label.rs`.

Reply:
The issue with using `Background` is that painting the background is a special case that is handled in `paint_view`. And that would require that the user always set a border radius. Which could be done in a default style?

Reply:
But that border radius needs be relative to the size of the element which isn't super convenient to set as a user